lib/collection/src/shards/replica_set/update.rs

```diff
@@ -185,13 +185,23 @@ impl ShardReplicaSet {
 
         let total_results = all_res.len();
 
+        let write_consistency_factor = self
+            .collection_config
+            .read()
+            .await
+            .params
+            .write_consistency_factor
+            .get() as usize;
+
+        let minimal_success_count = write_consistency_factor.min(total_results);
+
         let (successes, failures): (Vec<_>, Vec<_>) = all_res.into_iter().partition_result();
 
         // Notify consensus about failures if:
         // 1. There is at least one success, otherwise it might be a problem of sending node
         // 2. ???
 
-        if !successes.is_empty() {
+        if successes.len() >= minimal_success_count {
```

Comment:
Nitpick, but maybe we can assign this condition to a boolean variable, so we simplify our if statement conditions a bit.

```rust
let got_enough_successes = successes.len() >= minimal_success_count;

if got_enough_successes {}

if !failures.is_empty() && !got_enough_successes {}
```